dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -222,19 +220,9 @@ public TrackedEntity getTrackedEntity(
       }
     }
 
-    User currentUser = userService.getUserByUsername(CurrentUserUtil.getCurrentUsername());
-
     TrackedEntity trackedEntity;
     if (program != null) {
       trackedEntity = getTrackedEntity(uid, program, params, includeDeleted);
-
-      if (!trackerAccessManager.canRead(currentUser, trackedEntity, program, false).isEmpty()) {
-        if (program.getAccessLevel() == AccessLevel.CLOSED) {
-          throw new ForbiddenException(TrackerOwnershipManager.PROGRAM_ACCESS_CLOSED);
-        }
-        throw new ForbiddenException(TrackerOwnershipManager.OWNERSHIP_ACCESS_DENIED);
-      }
```

Comment:
I'm sorry you're removing this, but can you explain where we are going to check program access? Is this in the program service? so we don't do the acl check anymore?

Reply:
we do it in the same service, but inside the `getTrackedEntity` method now: https://github.com/dhis2/dhis2-core/pull/16783/files#diff-069b27403f4f267ecb5e14378a2e9c32d339e92acb494e688b821df93de33acbR274
I think it's a bit cleaner this way

Reply:
ah ok, but that is a private method then, and not implementing the interface